### PR TITLE
fix(linter): `useSortedClasses` false positive

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes https://github.com/biomejs/biome/issues/4041

## Test Plan

Added a new test

### CODE REVIEW

File: crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/sort.rs [hunk 1]
```diff
@@ -166,6 +166,7 @@ pub fn sort_class_name(
             }
         }
     }
+    dbg!(&sorted_classes, &classes_info);
```
Comment (dyc3) [inline crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/sort.rs:169]: nit: remove debug
```suggestion
```

File: crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/sort.rs [hunk 2]
```diff
@@ -207,6 +208,7 @@ pub fn sort_class_name(
         }
     }
 
+    dbg!(&result);
```
Comment (dyc3) [inline crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/sort.rs:211]: nit: remove debug
```suggestion
```

File: crates/biome_js_analyze/tests/quick_test.rs
Comment (dyc3) [inline crates/biome_js_analyze/tests/quick_test.rs:1]: revert changes to quick test

**Review by dyc3 — APPROVED** [review]

File: crates/biome_js_analyze/tests/quick_test.rs
Comment (ematipico) [reply crates/biome_js_analyze/tests/quick_test.rs:1]: That's intentional. We use `quick_test` across the board

File: crates/biome_js_analyze/tests/quick_test.rs [hunk 3]
```diff
@@ -73,6 +72,7 @@ fn analyze(
     let options = create_analyzer_options(input_file, &mut diagnostics);
     let manifest = load_manifest(input_file, &mut diagnostics);
 
+    dbg!("eree4");
```
Comment (ematipico) [inline crates/biome_js_analyze/tests/quick_test.rs:75]:
```suggestion
```

**Pull request merged**